examples/ShowPlayer.cpp

```diff
@@ -170,9 +173,9 @@ ShowLoader::State ShowPlayer::SeekTo(PlaybackTime seek_time) {
 void ShowPlayer::SendNextFrame() {
   ShowEntry entry;
   ShowLoader::State state = m_loader.NextEntry(&entry);
-  if (state == ShowLoader::END_OF_FILE ||
-      (m_stop > 0 && m_playback_pos >= m_stop)) {
-    if (entry.buffer.Size() > 0) {
+  const bool stop_point_hit = m_stop > 0 && m_playback_pos >= m_stop;
+  if (state == ShowLoader::END_OF_FILE || stop_point_hit) {
+    if (stop_point_hit) {
       // Send the last frame before looping/exiting
       SendFrame(entry);
     }
```

Comment:
I think it should only output this last frame if m_playback_pos == m_stop, otherwise you're potentially sending a frame after m_stop, which isn't what you've asked for in the CLI options.

Reply:
So stop_point_hit could now be just put directly in line `if (state == ShowLoader::END_OF_FILE || stop_point_hit) {` just above.